client.go

```diff
@@ -446,6 +450,18 @@ func DialWithTLS(network, address string, tlsConfig *tls.Config) (conn *Conn, er
 	return conn, nil
 }
 
+// Client.DialTimeoutEx acts like DialTimeout, but uses the client's settings.
```

Comment:
Not entirely sure what you are suggesting, but we are calling `client.Exchange` to perform a synchronous query. But we want to be in control of the local address. The changes in this PR address that. Not sure how to achieve that with `Dial` without having to copy a lot of code.